vector/src/androidTest/java/im/vector/app/ui/TrafficLightClientTest.kt

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2020 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.ui
+
+import android.Manifest
+import androidx.test.espresso.IdlingPolicies
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.rule.GrantPermissionRule
+import im.vector.app.espresso.tools.ScreenshotFailureRule
+import im.vector.app.features.MainActivity
+import im.vector.app.ui.robot.ElementRobot
+import okhttp3.MediaType
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import org.json.JSONObject
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.RuleChain
+import org.junit.runner.RunWith
+import java.lang.Thread.sleep
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+/**
+ * TrafficLight client is a UI test that receives requests from a TrafficLight server and fufils them on the UI
+ */
+@RunWith(AndroidJUnit4::class)
+class TrafficLightClientTest {
+
+    @get:Rule
+    val testRule: RuleChain = RuleChain
+            .outerRule(ActivityScenarioRule(MainActivity::class.java))
+            .around(GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE))
+            .around(ScreenshotFailureRule())
+
+    private val elementRobot = ElementRobot()
+
+    private val client = OkHttpClient()
+
+    @Test
+    fun trafficLightCycle() {
+        IdlingPolicies.setMasterPolicyTimeout(120, TimeUnit.SECONDS)
+
+        // keep the same UUID during each run; otherwise it should be entirely random.
+        val uuid = UUID.randomUUID().toString()
+        val trafficLightServer = "http://10.0.2.2:5000"
+        val postURL = "$trafficLightServer/client/$uuid/respond"
+        val pollURL = "$trafficLightServer/client/$uuid/poll"
+        val registerURL = "$trafficLightServer/client/$uuid/register"
+
+        register(registerURL)
+        while (true) {
```

Comment:
to double check, are we using the `setMasterPolicyTimeout` as our safety net to avoid the test running forever?

eg if no response is received from the traffic light server for 2mins+, will the test case end?